core/src/test/java/org/apache/iceberg/hadoop/TestCachingCatalog.java

```diff
@@ -166,6 +168,42 @@ public void testTableName() throws Exception {
         .isEqualTo("hadoop.db.ns1.ns2.tbl.snapshots");
   }
 
+  @Test
+  public void testNonExistingTable() throws Exception {
+    Catalog catalog = CachingCatalog.wrap(hadoopCatalog());
+
+    TableIdentifier tableIdent = TableIdentifier.of("otherDB", "otherTbl");
+
+    assertThatThrownBy(() -> catalog.loadTable(tableIdent))
+        .isInstanceOf(NoSuchTableException.class);
```

Comment:
please also add `.hasMessage("Table does not exist: otherDB.otherTbl")`here as we try to avoid to only check for the exception and typically always want to make sure we get the right error msg

Reply:
Sure, done